packages/eslint-plugin/src/rules/shorthand-property-sorting/__tests__/rule.test.ts

```diff
@@ -0,0 +1,240 @@
+tester.run('shorthand-property-sorting', shorthandFirst, {
+  valid: [
+    ...packages_calls_and_imports.map(([pkg, call, imp]) => ({
+      name: `correct property ordering, (${pkg}: '${imp}')`,
+      code: outdent`
+      import {${pkg}} from '${imp}';
+      const styles = ${call}({
+        margin: '1', // 1
+        border: '2', // 1
+        borderColor: '7', // 2
+        borderBlock: '3', // 3
+        borderBottom: '6', // 4
+        borderBlockEnd: '4', // 5
+        borderBlockStart: '5', // 5
+      });
+      export const EmphasisText1 = ({ children }) => <span css={styles}>{children}</span>;
+    `,
+    })),
+    ...packages_and_calls.map(([pkg, call]) => ({
+      name: `incorrect property ordering, (${pkg}: from unregulated package)`,
+      code: outdent`
+      import {${pkg}} from 'wrongpackage';
+      const styles = ${call}({
+        borderTop: '1px solid #00b8d9',
+        border: '#00b8d9',
+      });
+      export const EmphasisText = ({ children }) => <span css={styles}>{children}</span>;
+    `,
+    })),
+    ...packages_calls_and_imports.map(([pkg, call, imp]) => ({
+      name: `properties that don't interact with out of order depths (${pkg}: '${imp}')`,
+      code: outdent`
+      import {${pkg}} from '${imp}';
+      const styles = ${call}({
+        borderColor: '#00b8d9', // 2
+        font: '#00b8d9', // 1
+      });
+      export const EmphasisText = ({ children }) => <span css={styles}>{children}</span>;
+    `,
+    })),
+    ...packages_calls_and_imports.map(([pkg, call, imp]) => ({
+      name: `property not in bucket (${pkg}: '${imp}'`,
+      code: outdent`
+      import {${pkg}} from '${imp}';
+      const styles = ${call}({
+        transitionDuration: '2', // unknown
+        transition: 'fast', // 1
+      });
+      export const EmphasisText = ({ children }) => <span css={styles}>{children}</span>;
+    `,
+    })),
+    ...packages_calls_and_imports.map(([pkg, call, imp]) => ({
+      name: `depth in correct order for properties in the same bucket (${pkg}: '${imp}')`,
+      code: outdent`
+          import {${pkg}} from '${imp}';
+          const styles = ${call}({
+            borderColor: '#00b8d9', // 2
+            borderTop: '1px solid #00b8d9', // 4
+          });
+          export const EmphasisText = ({ children }) => <span css={styles}>{children}</span>;
+        `,
+    })),
+  ],
+  invalid: [
+    ...packages_calls_and_imports.map(([pkg, call, imp]) => ({
+      name: `incorrect property ordering (${pkg}: '${imp}')`,
+      code: outdent`
+      import {${pkg}} from '${imp}';
+      const styles = ${call}({
+        borderTop: '1px solid #00b8d9',
+        border: '#00b8d9',
+      });
+      export const EmphasisText = ({ children }) => <span css={styles}>{children}</span>;
+    `,
+      errors: [{ messageId: 'shorthand-first' }],
+    })),
+    ...packages_calls_and_imports.map(([pkg, call, imp]) => ({
+      name: `incorrect property ordering -> 3 properties (${pkg}: '${imp}'`,
```

Comment:
out of curiosity, what information does `3 properties` provide?

Reply:
Not a whole lot since the rest of the tests also deal with this, can take it out while I make the other changes.